packages/eslint-plugin/lib/rules/no-nested-await.js

```diff
@@ -45,6 +45,24 @@ module.exports = {
     supported: true || CodePathAnalyzer !== null,
   },
   create(context) {
+    const isValidParent = parent => {
+      if (parent.type === 'BlockStatement') {
+        // Find the parent block's node.
+        parent = parent.parent;
+      }
+
+      // Is the parent is an async function or for-await-of body.
```

Comment:
IIUC "for-await-of" is both an "await" and allowed as a parent of other "await"s, like so:

```js
async () => {
  for await (const e1 of iter1) {
    await something;
    for await (const e2 of iter2) {
      await something else;
    }
  }
};
```

Here is a suggestion for allowing code like the above:

```suggestion
      // Is the parent a for-await-of?
      if (parent.type === 'ForOfStatement' && parent.await === true) {
        // Is the await allowed by the parent of the for-await-of?
        return isAwaitableParent(parent.parent);
      }
      // Is the parent an async function?
      return [
        'ArrowFunctionExpression',
        'FunctionExpression',
        'FunctionDeclaration',
      ].includes(parent.type) && parent.async;
```

Reply:
I have verified that the PR as-is currently allows nested `for-await-of` statements, both with a block as the body, or directly with a statement. The above change is unnecessary since the current check basically causes a `for-await-of` block to form a valid "top level scope" the same way an async function would. What makes you think this change is necessary?

Reply:
> the current check basically causes a `for-await-of` block to form a valid "top level scope" the same way an async function would.

Ah, okay.  I misunderstood.